src/js/htmlSanitizer.js

```diff
@@ -28,6 +28,10 @@ const SVG_ATTR_LIST_RX = new RegExp('^(accent-height|accumulate|additive|alphabe
     'width|widths|x|x-height|x1|x2|xlink:actuate|xlink:arcrole|xlink:role|xlink:show|xlink:title|' +
     'xlink:type|xml:base|xml:lang|xml:space|xmlns|xmlns:xlink|y|y1|y2|zoomAndPan)', 'g');
 
+const ATTR_VALUE_BLACK_LIST_RX = {
+  'href': new RegExp('^(javascript:).*', 'g')
+};
```

Comment:
I think you'll have to deal with all three of the following cases.
```html
<object data='data:text/html;base64,PHN2Zy9vbmxvYWQ9YWxlcnQoIlhTUyIpPg=='></object>
<embed src="data:text/html;base64,PHN2Zy9vbmxvYWQ9YWxlcnQoIlhTUyIpPg=="></embed>
<a href=javascript:alert(document.domain)>hello</a>
```

Reply:
This also works

```html
<object data='data:text/html;base64,PGltZyBzcmM9Ii8xLnBuZyIgb25lcnJvcj1hbGVydCgiWFNTIik+'></object>
```
**base64 decoded**
```html
PGltZyBzcmM9Ii8xLnBuZyIgb25lcnJvcj1hbGVydCgiWFNTIik+
<img src="/1.png" onerror=alert("XSS")>
```

It would be more robust if `object` and `embed` also sanitize element they are rendering, if it is `html`.

Reply:
@LunaTK
You're right. It's missing from the code. You can add that tag [here](https://github.com/nhn/tui.editor/blob/master/src/js/htmlSanitizer.js#L57).

Reply:
@seonim-ryu 
I checked [Github's Specification](https://github.github.com/gfm/#html-block) about rendering raw HTML to see how Github renders `embed` and `object` tags.

Github only treats following tags as valid tag. Otherwise, it seems renderer just **unwrap the invalid tag** and render child elements.

> Start condition: line begins the string `<` or `</` followed by one of the strings (case-insensitive) `address`, `article`, `aside`, `base`, `basefont`, `blockquote`, `body`, `caption`, `center`, `col`, `colgroup`, `dd`, `details`, `dialog`, `dir`, `div`, `dl`, `dt`, `fieldset`, `figcaption`, `figure`, `footer`, `form`, `frame`, `frameset`, `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `head`, `header`, `hr`, `html`, `iframe`, `legend`, `li`, `link`, `main`, `menu`, `menuitem`, `nav`, `noframes`, `ol`, `optgroup`, `option`, `p`, `param`, `section`, `source`, `summary`, `table`, `tbody`, `td`, `tfoot`, `th`, `thead`, `title`, `tr`, `track`, `ul`, followed by whitespace, the end of the line, the string `>`, or the string `/>`.


To summarize, **[GFM](https://github.github.com/gfm/#raw-html) does not allow** `embed` and `object` tags to be rendered.

Since `tui-editor` claims that it follows GFM, **unwrapping** invalid HTML tags seems to be more appropriate.

However, current policy of `tui-editor`'s sanitizer for invalid tags is to **remove** them, including their child elements.
https://github.com/nhn/tui.editor/blob/01e83e63ddeb5ab7987027513c49be5319773d82/src/js/htmlSanitizer.js#L38-L49


So what do you want me to do?
1. Unwrap
2. Remove


> @LunaTK
> You're right. It's missing from the code. You can add that tags [here](https://github.com/nhn/tui.editor/blob/master/src/js/htmlSanitizer.js#L57).


Do you want to `remove` them?

Reply:
@LunaTK
Yes, **remove** them.

Once our basic policy is to follow GFM. However, depending on the situation, the spec is changed and used. According to the current editor's spec, it should be removed.

Issue that follow the GFM specification you mentioned will be registered as features and reviewed later.